JSON Attribute + Übersetzungen + dark mode fixed

skerbis opened the pull request:
# Konfigurationsformat für YForm Geo OSM vereinfacht und Dark Mode verbessert

Mit diesem PR wird das Konfigurationsformat für die Kartenansicht vereinfacht und der Dark Mode grundlegend überarbeitet.

## Änderungen

### Neues Format- und Konfigurationskonzept
- Einführung eines einfachen JSON-Formats für Karteneinstellungen 
  ```json
  {
      "style": "height: 400px",
      "data-init-lat": "52.5200",
      "data-init-lng": "13.4050",
      "data-init-zoom": "12"
  }
  ```
- Kompatibilität mit bestehender height-Option beibehalten

### Dark Mode und CSS
- Optimierte Kartendarstellung für Dark Mode (bessere Lesbarkeit, angepasste Kontraste)
- Modernere Transitions und Hover-Effekte
- Konsistenteres Look & Feel im Backend

### User Experience
- Marker wird erst bei Positionsauswahl angezeigt
- Vorauswahl der Position über data-Attribute möglich
- Verbesserte Hover-Zustände für Buttons und Interaktionselemente

## Migration

### Vor der Änderung:
```php
$yform->setValueField('osm_geocode', ['osm','OSM','lat,lng','street,postalcode,city','400']);
```

### Nach der Änderung:
```php
$yform->setValueField('osm_geocode', ['osm','OSM','lat,lng','street,postalcode,city','{"style":"height: 400px"}']);
```

### Optionale Konfiguration:
```php
// Mit initialer Position
$yform->setValueField('osm_geocode', ['osm','OSM','lat,lng','street,postalcode,city','{"style":"height: 400px","data-init-lat":"52.5200","data-init-lng":"13.4050","data-init-zoom":"12"}']);

// Mit zusätzlichen Klassen
$yform->setValueField('osm_geocode', ['osm','OSM','lat,lng','street,postalcode,city','{"style":"height: 400px","class":"my-custom-class"}']);
```

## Breaking Changes
- Marker wird initial nicht mehr angezeigt (erscheint erst bei Positionsauswahl).

## Getestete Szenarien
- Migration bestehender Felder (height-Option)
- Dark Mode Darstellung in verschiedenen Browsern
- Marker-Verhalten bei Positionsauswahl
- Initiale Kartenpositionen über data-Attribute
- CSS-Variablen im Light/Dark Mode


<img width="1029" alt="Bildschirmfoto 2025-01-02 um 21 19 27" src="https://github.com/user-attachments/assets/9335aad1-6296-4384-9c02-8b9c547f98ee" />

Alle Texte aus dem Value wurden in lng-dateien verschoben. Wie ich Übersetzungen für den Geopicker (für Module) erstelle weiß ich noch nicht. 

Dark-Mode wurde auch verbessert

<img width="1188" alt="Bildschirmfoto 2025-01-02 um 22 01 41" src="https://github.com/user-attachments/assets/19034a92-1ab1-43b1-a276-b8f02cee6838" />

<img width="943" alt="Bildschirmfoto 2025-01-02 um 22 01 31" src="https://github.com/user-attachments/assets/2b9aee8e-ede2-4b2e-94e8-49ba0542bbb8" />


fixes: https://github.com/FriendsOfREDAXO/yform_geo_osm/issues/41
fixes: https://github.com/FriendsOfREDAXO/yform_geo_osm/issues/37


👋 jetzt erstmal raus hier ... viel Spaß noch damit

skerbis committed:
Update de_de.lang

skerbis committed:
Update en_gb.lang

skerbis committed:
Update osm_geocode.php

skerbis committed:
Update value.osm_geocode.tpl.php

skerbis committed:
Update en_gb.lang

skerbis committed:
Update de_de.lang

skerbis committed:
Update geo_osm.js

skerbis committed:
Update geo_osm.js

skerbis committed:
Update geo_osm.js

skerbis committed:
Update README.md

skerbis committed:
Update README.md

skerbis committed:
Update README.md

skerbis committed:
dark mode fixed

skerbis committed:
Update CHANGELOG.md

skerbis committed:
Update CHANGELOG.md

AWqxKAWERbXo reviewed (commented):
Für mich geht das klar

skerbis committed:
Update geo_osm.js

Hier ist die Pull Request Beschreibung für die JavaScript-Optimierung:

# Verbesserung der OSM-Geocoding Funktionalität

## 🎯 Änderungen

### Verbesserte Adressfeld-Verarbeitung
- Flexible Verarbeitung der kommaseparierten Adressfelder ohne feste Reihenfolge
- Robuste Filterung leerer Werte
- Bessere Fehlerbehandlung bei unvollständigen Adressen
- Korrekte Verarbeitung beliebig vieler Adressfelder (nicht mehr auf 3 Felder beschränkt)

christophboecker commented:
Ich schau es mir an, muss aber erstmal ne Testumgebung bauen

christophboecker reviewed (approved):
Also: 
- das neue Feld "Format" sollte m.E. hinter "Map-Höhe" kommen. Dann sind alle neuen Felder hinten dran und es entsteht kein BC mit viel Anpassungsaufwand. (DX)
- Übergabe der Daten an das HTML sollte man, danke für die Top-Idee, via HTML-Attribute übergeben und nicht per JS-Funktion. Dann aber ALLE, auch die Position und die Feldnamen und so.
- Ich denke nicht, dass es nötig ist, für jedes Topic ein eigenes Data-Attribut zu verwenden; ein bis zwei (data-config="{cord=latlng, address=plz,ort,strasse, lat=,lng=,zoom=,class=,height=} data-coord={lat=,lng=}") sollten reichen. Dann kann übrigens auch die imho unleidige JS-Initialisierung samt OutputFilter-Einbau entfallen.

AWqxKAWERbXo commented:
Ich würde gerne ein eigenes Data-Attribut sehen, um die Lesbarkeit zu erhöhen. Verschachtelte Attribute machen mich bereits beim Upload-Feld fertig.
